plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -2260,16 +2262,25 @@ private void pruneAndDeleteStagingDirectories(List<DeclaredIntentionToWrite> dec
             }
         }
 
-        private void waitForAsyncRenames()
+        private void waitForAsyncFileSystemOperations()
         {
-            for (CompletableFuture<?> fileRenameFuture : fileRenameFutures) {
-                getFutureValue(fileRenameFuture, TrinoException.class);
+            try {
+                fileSystemFutures.forEach(future -> getFutureValue(future, TrinoException.class));
```

Comment:
seems like `getFutureValue(future)` can be used as `TrinoException` is already `RuntimeException`